protocol/proofs/anchor-state-poison-protection.md

```diff
@@ -0,0 +1,121 @@
+anchor states in other game types.
+
+Our problem statement here is therefore that we wish to be able to prevent or mitigate anchor state
+poisoning more effectively so that we can reduce the surface for issues within the fault proof
+system as a whole.
+
+## Proposed Solution
+
+We propose solving this problem by updating the `AnchorStateRegistry` to more generally be the
+source of truth for the validity of states from the perspective of the L1 proof system. All
+functionality in the `OptimismPortal` related to state validity would be moved to the
+`AnchorStateRegistry` contract. This proposed solution would simplify the `OptimismPortal`
+contract, clarify the concept of state validity from the PoV of the proof system, and solve the
+state poisoning issue within the `AnchorStateRegistry`.
+
+### Changes to the AnchorStateRegistry
+
+The `AnchorStateRegistry` would be modified to take over the blacklisting and air-gap roles that
+are currently handled by the `OptimismPortal` contract. The `AnchorStateRegistry` would expose
+several new getter functions that can be used by the `OptimismPortal` contract to determine if a
+given dispute game can be permitted to finalize a withdrawal.
+
+The `AnchorStateRegistry` would continue to keep track of the results of `FaultDisputeGame`
+contracts via the current "push" model where the `FaultDisputeGame` attempts to trigger an update
+in the `AnchorStateRegistry` on game finalization. However, instead of immediately updating the
+anchor state when a game is finalized, the `AnchorStateRegistry` would instead store the result in
+a log of game results. Game results would only be promoted to anchor states if they've passed the
+air-gap window without being blacklisted.
+
+We can avoid making significant changes to the current challenger setup by using the call to the
+`tryUpdateAnchorState` function that the `FaultDisputeGame` already makes to both store the game
+result in the game array and to use a small binary search to update the anchor state. A binary
+search is likely to be acceptable here because we can bound the starting end of the search range to
+the index of the current anchor state and only search over states that finalized after the current
+anchor state was finalized. Although this can potentially miss a few eligible anchor states under
+certain circumstances, this is an acceptable trade-off as new anchor states *will* eventually be
+found after the current anchor state.
+
+### Changes to the OptimismPortal
+
+State validity rules like blacklisting and air-gap window inside of the `OptimismPortal` would be
+removed and outsourced to the `AnchorStateRegistry`. New public getter functions in the
+`AnchorStateRegistry` would provide the information that the `OptimismPortal` needs to determine
+whether a given dispute game contract can be used to finalize a withdrawal.
+
+### Changes to the FaultDisputeGame
+
+No changes to the FaultDisputeGame contract would be required.
+
```

Comment:
As we will need to upgrade the `DGM` Should we add a part into the current doc to also informed about this breaking changes for future readers?

```suggestion
### Changes to the DeputyGuardianModule
The DeputyGuardianModule will have few light changes but this would require a total redeployment on L1. Thus, breaking the current PSPs and would need to regenerate them. 
```

Reply:
Good point, thank you.